### PR TITLE
FIREWALL: Allow to block hosts by IP address

### PR DESCRIPTION
**Pull request by aplopez**
When blocking a host, its hostname is resolved using the `dig` command. If an IP address is provided, `dig` returns nothing and it is not blocked.

We need to check whether it is an IP address before launching `dig`. No matter whether the correct family is requested, do not call `dig` when an IP address is provided, in case the caller has already blocked the DNS host.

### CODE REVIEW

**Comment by aplopez** [comment]
tox failures existed before this PR.

**Comment by pbrezina** [comment]
Hi, please rebase to get CI green.

**Comment by aplopez** [comment]
> Hi, please rebase to get CI green.

Perfect. Thank you.

File: pytest_mh/utils/firewall.py [hunk 1]
```diff
@@ -605,9 +606,28 @@ def __add_host(
...
+            ipv6 = None
+
+        if ipv4 is not None or ipv6 is not None:
+            # In this case we were given an IP address but of the wrong family. We must not consider it a hostname.
+            return []
+        else:
+            result = self.firewall.host.conn.exec(
+                ["dig", "+short", "-t", type, hostname], log_level=ProcessLogLevel.Error
+            )
+            return result.stdout_lines
```
Comment (pbrezina) [inline pytest_mh/utils/firewall.py:630]: How about simplifying it to:

```python
addrs = []
try:
    ip = ipaddress.ip_address(hostname)
    ip_type = ipaddress.IPv4Address if type == "A" else ipaddress.IPv6Address
    if isinstance(ip, ip_type):
        addrs = [hostname]
except ValueError:
    result = self.firewall.host.conn.exec(
        ["dig", "+short", "-t", type, hostname], log_level=ProcessLogLevel.Error
    )
   addrs = result.stdout_lines

return addrs
```

**Review by pbrezina — APPROVED** [review]
Ack. Thank you. Since I contributed to the solution, somebody else should review it as well.

**Review by justin-stephenson — APPROVED** [review]

**Pull request merged**